### PR TITLE
checksum

### PR DESCRIPTION
**Pull request by kaisbaccour**
https://github.com/Lagrange-Labs/zkmr-tracking/issues/109

### CODE REVIEW

File: lgn-provers/src/params/mod.rs [hunk 1]
```diff
@@ -136,4 +157,34 @@ impl ParamsLoader {
...
+            .context("Failed to read file for checksum verification")?;
+        hasher.update(&buffer);
+        let actual_checksum = format!("{:x}", hasher.finalize());
+        if actual_checksum != expected_checksum {
+            info!(
+                "Checksum mismatch: expected {}, but got {}",
+                expected_checksum, actual_checksum
+            );
+        }
+        Ok(actual_checksum == expected_checksum)
```
Comment (nikkolasg) [inline lgn-provers/src/params/mod.rs:175]: Mhhh can we maybe use standard tools for this ? it's gonna be easier to port accross platforms and easier to write scripts as well in bash. Like `sha256sum` or `shasum` i think. Also probably might be a bit faster.

Comment (nikkolasg) [reply lgn-provers/src/params/mod.rs:175]: But in this case, we would have to download + save before checking checksum.
Which is fine for me btw.

Another solution would be to [use blake3](https://docs.rs/blake3/latest/blake3/struct.Hasher.html#method.update_rayon) which supports reading from mmap and even do stuff in parallel with rayon and blake3 is known to be much faster, even more for large files.

File: lgn-provers/src/provers/v0/groth16/mod.rs [hunk 2]
```diff
@@ -9,12 +9,16 @@ mod prover;
 mod task;
 
 #[allow(unused_variables)]
+#[allow(clippy::too_many_arguments)]
 pub fn create_prover(
     url: &str,
     dir: &str,
     circuit_file: &str,
+    circuit_file_checksum: &str,
```
Comment (nikkolasg) [inline lgn-provers/src/provers/v0/groth16/mod.rs:17]: Can't we simply do the checksum over the whole directory maybe ? Simpler ? It's best to be atomic on this stuff, we don't want to have different versions of the sub-parameters living the same place.

File: lgn-provers/src/provers/v0/query/mod.rs [hunk 3]
```diff
@@ -11,6 +11,7 @@ pub fn create_prover(
     url: &str,
     dir: &str,
     file: &str,
+    checksum: &str,
```
Comment (nikkolasg) [inline lgn-provers/src/provers/v0/query/mod.rs:14]: same, can't we do it simply over the whole parameter directory and be done with it ? The individual provers shouldn't have to know about this, they should expect to already get the right version.

**Review by nikkolasg — COMMENTED** [review]
TLDR:
* Can we use standard tools (easier to write scripts as well on our side), and if not, something better than sha2
* Can we have a _single_ checksum for the whole directory.
